gtdynamics/universal_robot/Robot.h

```diff
@@ -156,24 +156,26 @@ class Robot {
    */
   gtsam::Values forwardKinematics(
       const gtsam::Values &known_values, size_t t = 0,
-      const boost::optional<std::string> &prior_link_name = boost::none) const;
+      const std::optional<std::string> &prior_link_name = {}) const;
 
  private:
   /// Find root link for forward kinematics
   LinkSharedPtr findRootLink(
       const gtsam::Values &values,
-      const boost::optional<std::string> &prior_link_name) const;
+      const std::optional<std::string> &prior_link_name) const;
```

Comment:
should this have a default argument?  or is it strict that user should explicitly specify {} if they don't want to provide it?

Reply:
This is a helper function, which is only called in `forwardKinematics` that always pass in the optional typed variable.

Personally I think it's fine to not provide a default argument here.